app/src/main/java/org/stepik/android/view/step_quiz/ui/delegate/StepQuizDelegate.kt

```diff
@@ -182,5 +183,11 @@ class StepQuizDelegate(
             ?: return
 
         onNewMessage(StepQuizFeature.Message.SyncReply(reply))
+
+        if (step.block?.name == AppConstants.TYPE_CODE) {
+            val languagesKey = step.block?.options?.codeTemplates?.keys?.toList()?.sorted() ?: return
+            val preferredLanguage = reply.language ?: return
+            onNewMessage(StepQuizFeature.Message.CreateCodePreference(languagesKey = languagesKey.toString(), language = preferredLanguage))
```

Comment:
it's strange that this message emitted not in `CodeLangAdapterDelegate`'s onclick